pkg/deploy/deploy.go

```diff
@@ -287,6 +287,11 @@ func manageResource(ctx context.Context, cli client.Client, obj *unstructured.Un
 		return nil
 	}
 
+	if found.GetAnnotations()["opendatahub.io/managed"] == "false" {
```

Comment:
The other patch is responsible for removing the ownerreferrences and cleaning up the resources when a component is set to `Removed` . I think we still want to cleanup the resource with "opendatahub.io/managed" annotation...correct?

If so, we don't need to update any condition.

Reply:
Is there any particular reason for `"opendatahub.io/managed"` to be an annotation instead of a label? With the  latter, we could query such resources easier.  Can the purpose of this be documented as part of this PR? This could help us decide what's the most feasible approach here. 

Also if it's not documented it becomes a super-power known only by chosen ones :)

Reply:
just found out label  `opendatahub.io/managed` has been used by several components already.
so if we want to make it a label than annotation, we need to find a different key for it.